prow/tide/tide.go

```diff
@@ -108,47 +124,60 @@ type Pool struct {
 }
 
 // NewController makes a Controller out of the given clients.
-func NewController(ghc *github.Client, kc *kube.Client, ca *config.Agent, gc *git.Client, logger *logrus.Entry) *Controller {
+func NewController(ghcSync, ghcStatus *github.Client, kc *kube.Client, ca *config.Agent, gc *git.Client, logger *logrus.Entry) *Controller {
+	sc := &statusController{
+		logger:         logger.WithField("controller", "status-update"),
+		ghc:            ghcStatus,
+		ca:             ca,
+		newPoolPending: make(chan bool, 1),
+	}
+	go sc.run()
```

Comment:
How is this going to manage shutdown? What happens when it is killed in the middle of updating a pool?

Reply:
Seems that it is reentrant so this is probably ok.